wire-protoc-compatibility-tests/src/test/java/com/squareup/wire/Proto3WireProtocCompatibilityTests.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+import com.google.protobuf.DescriptorProtos
+import com.google.protobuf.FieldOptions
+import com.squareup.wire.proto3.requiredextension.RequiredExtension
+import com.squareup.wire.proto3.requiredextension.RequiredExtensionMessage
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class Proto3WireProtocCompatibilityTests {
+  // Note: this test mostly make sure we compile required extension without failing.
+  @Test fun protocAndRequiredExtensions() {
+    val wireMessage = RequiredExtensionMessage("Yo")
+
+    val googleMessage = RequiredExtension.RequiredExtensionMessage.newBuilder()
+        .setStringField("Yo")
+        .build()
+
+    assertThat(wireMessage.encode()).isEqualTo(googleMessage.toByteArray())
+
+    // Although the custom options has no label, it shouldn't be "required" to instantiate
+    // `FieldOptions`.
+    DescriptorProtos.FieldOptions.newBuilder().build()
```

Comment:
Missing asserts here? If expected behavior is "no exception thrown" then try/catch and `fail()` inside the catch block.

Reply:
I also use `assertNotNull` for cases like this with a comment.

Reply:
went with that. the try/catch adds nesting, and many lines.
Thanks to you both